### PR TITLE
Adding "CiCo: Domain-Aware Sign Language Retrieval via Cross-Lingual Contrastive Learning"

### PR DESCRIPTION
**Pull request by cleong110**
Paper: https://openaccess.thecvf.com/content/CVPR2023/html/Bao_CiCo_Domain-Aware_Sign_Language_Retrieval_via_Cross-Lingual_Contrastive_Learning_CVPR_2023_paper.html

Added a summary of the paper, as well as some of the works leading up to it. And also citations to a few related papers. 

https://github.com/cleong110/sign-language-processing.github.io/issues/16 my writing process. 

TODO still: 
- [x] another editing pass with ChatGPT-assistance for conciseness. 
- [ ] Code and models are available at: https://github.com/FangyunWei/SLRT, maybe add a link.

### CODE REVIEW

File: src/index.md [hunk 1]
```diff
@@ -902,8 +902,16 @@ TODO
...
-<!-- TODO: text-to-sign-video (T2V) section, sign-video-to-text (V2T) retrieval -->
-<!-- TODO: CiCo: Domain-Aware Sign Language Retrieval via Cross-Lingual Contrastive Learning -->
+Metrics used include retrieval at Rank K and (R@K, higher is better) and median rank (MedR, lower is better).
+
+<!-- TODO: text-to-sign-video (T2V) section, sign-video-to-text (V2T) retrieval? -->
+@athitsos2010LargeLexiconIndexingRetrieval present one of the early works in this task, using a method based on hand centroids and dynamic time warping to enable users to submit videos of a sign and thus query within the ASL Lexicon Video Dataset [@dataset:athitsos2008american].
+
+@Zhang2010RevisedEditDistanceSignVideoRetrieval provide another early method for video-based querying.
+They use classical image feature extraction methods to calculate movement trajectories.
+They then use modified string edit distances between these trajectories as a way to find similar videos.
```
Comment (AmitMY) [inline src/index.md:912]: very good!

File: src/index.md [hunk 2]
```diff
@@ -912,6 +920,17 @@ When a user submits a query video, the system compares the input embeddings with
 Tests on a [proof-of-concept Flemish Sign Language dictionary](https://github.com/m-decoster/VGT-SL-Dictionary) show that the system can successfully retrieve a limited vocabulary of signs, including some not in the training set.
 <!-- TODO: add VGT Corpus (dataset:herreweghe2015VGTCorpus) to list of datasets -->
 
+<!-- TODO: Sign language video retrieval with free-form textual queries was the only other paper that Cheng2023CiCoSignLanguageRetrieval compared with. -->
+
+@Cheng2023CiCoSignLanguageRetrieval introduce a video-to-text (V2T) and text-to-video (t2V) retrieval method based on cross-lingual contrastive learning.
+Using a "domain-agnostic" I3D encoder pretrained on large-scale sign datasets [@Varol2021ReadAndAttend] they generate pseudo-labels on target datasets and finetune a "domain-aware" encoder.
+Combining the two encoders they then pre-extract features from sign language videos.
+They then use cross-lingual contrastive learning [@Radford2021LearningTV] in order to contrast feature/text pairs, mapping them to a shared embedding space.
```
Comment (AmitMY) [inline src/index.md:928]: you use "they then" 3 times in this PR, which makes the reading less "fun". chatgpt can help

**Review by AmitMY — CHANGES_REQUESTED** [review]

File: src/index.md [hunk 3]
```diff
@@ -912,6 +920,17 @@ When a user submits a query video, the system compares the input embeddings with
...
+Embeddings of matched pairs are pulled together and non-matched pairs pushed apart.
+They evaluate on How2Sign [@dataset:duarte2020how2sign] and RWTH-PHOENIX-Weather 2014T dataset [@cihan2018neural], improving by a substantial portion over the previous state of the art method [@Duarte2022SignVideoRetrivalWithTextQueries].
```
[the remark this answers is not in the document]
Comment (cleong110) [reply src/index.md:930]: Oh good point

File: src/index.md [hunk 4]
```diff
@@ -902,8 +902,16 @@ TODO
 
 Sign Language Retrieval is the task of finding a particular data item, given some input. In contrast to translation, generation or production tasks, there can exist a correct corresponding piece of data already, and the task is to find it out of many, if it exists.
 
-<!-- TODO: text-to-sign-video (T2V) section, sign-video-to-text (V2T) retrieval -->
-<!-- TODO: CiCo: Domain-Aware Sign Language Retrieval via Cross-Lingual Contrastive Learning -->
+Metrics used include retrieval at Rank K and (R@K, higher is better) and median rank (MedR, lower is better).
+
+<!-- TODO: text-to-sign-video (T2V) section, sign-video-to-text (V2T) retrieval? -->
+@athitsos2010LargeLexiconIndexingRetrieval present one of the early works in this task, using a method based on hand centroids and dynamic time warping to enable users to submit videos of a sign and thus query within the ASL Lexicon Video Dataset [@dataset:athitsos2008american].
```
[the remark this answers is not in the document]
Comment (cleong110) [reply src/index.md:908]: lol it is not! I think I crossed "on this task" with "in this field" or similar

File: src/index.md [hunk 5]
```diff
@@ -912,6 +920,17 @@ When a user submits a query video, the system compares the input embeddings with
 Tests on a [proof-of-concept Flemish Sign Language dictionary](https://github.com/m-decoster/VGT-SL-Dictionary) show that the system can successfully retrieve a limited vocabulary of signs, including some not in the training set.
 <!-- TODO: add VGT Corpus (dataset:herreweghe2015VGTCorpus) to list of datasets -->
 
+<!-- TODO: Sign language video retrieval with free-form textual queries was the only other paper that Cheng2023CiCoSignLanguageRetrieval compared with. -->
+
+@Cheng2023CiCoSignLanguageRetrieval introduce a video-to-text (V2T) and text-to-video (t2V) retrieval method based on cross-lingual contrastive learning.
```
[the remark this answers is not in the document]
Comment (cleong110) [reply src/index.md:925]: Good point

File: src/index.md (hunk 1 above)
Comment (cleong110) [reply src/index.md:912]: Thanks!!

File: src/index.md [hunk 6]
```diff
@@ -912,6 +920,17 @@ When a user submits a query video, the system compares the input embeddings with
...
+Using a "domain-agnostic" I3D encoder pretrained on large-scale sign datasets [@Varol2021ReadAndAttend] they generate pseudo-labels on target datasets and finetune a "domain-aware" encoder.
+Combining the two encoders they then pre-extract features from sign language videos.
```
[the remark this answers is not in the document]
Comment (cleong110) [reply src/index.md:927]: I don't think I fully understand either, tbh? 


Best I can gather they...

1. Start with a sign-spotting model from https://www.robots.ox.ac.uk/~vgg/research/bslattend/
> In this work, pseudo-labeling is served as our sign spotting approach to localize isolated signs in untrimmed videos from target sign language retrieval datasets.
>Previous methods [16, 18] **have demonstrated the feasibility of transferring a sign encoder pre-trained on large-scale sign-spotting data into downstream tasks.** We follow this practice and use an I3D network pre-trained on BSL-1K [62], a sign classification dataset collected via sign spotting, as our primary sign encoder. Due to its favorable transfer performance, we term this model as a domain-agnostic sign encoder.

2. For target datasets, finetune a target-dataset-specific sign-spotter? Basically take the high-confidence predictions as labels? I guess because the datasets they're interested don't have timestamps/ are not "sign-spotting" or isolated datasets. Thus the sliding window and the high-confidence predictions
image

3. use both at once in a sort of ensemble:
image

[the remark this answers is not in the document]
Comment (cleong110) [reply src/index.md:927]: OK, now that I think aloud about this, on say CSL-Daily that makes sense. 
1. Start with your BSL-1K sign-spotter
2. Run a sliding window acros CSL-Daily videos, pull out the really high-confidence ones
3. Use that to train a CSL-Daily sign-spotter.
4. Use both

**Comment by cleong110** [comment]
Something else I discovered: a number of citations go to 

```
@dataset:huang2018video
```

which _is_apparently a CSL dataset, but not the _CSL-Daily_ dataset

**Comment by cleong110** [comment]
Rewrote with ChatGPT assistance: https://chatgpt.com/share/007fb06b-e6df-468c-a7c3-9c8cc24b9855

**Comment by cleong110** [comment]
ChatGPT helped ID misformatted citations, and gave some helpful suggestions. I asked it this time to go line by line explaining changes, some of which I took and some I did not, or adjusted. For example it tried to but T2V and V2T back in

File: src/index.md (hunk 2 above)
Comment (cleong110) [reply src/index.md]: So I do! Fixed

File: src/index.md [hunk 7]
```diff
@@ -901,17 +901,34 @@ TODO
 
 ### Sign Language Retrieval
 
-Sign Language Retrieval is the task of finding a particular data item, given some input. In contrast to translation, generation or production tasks, there can exist a correct corresponding piece of data already, and the task is to find it out of many, if it exists.
+Sign Language Retrieval is the task of finding a particular data item, given some input. In contrast to translation, generation or production tasks, there can exist a correct corresponding piece of data already, and the task is to find it out of many, if it exists. Metrics used include retrieval at Rank K and (R@K, higher is better) and median rank (MedR, lower is better).
```
Comment (AmitMY) [inline src/index.md:904]: Remove wrong "and"

Metrics used include retrieval at Rank K (R@K, higher is better) and median rank (MedR, lower is better).

File: src/references.bib [hunk 8]
```diff
@@ -3261,3 +3278,78 @@ @inproceedings{dataset:starner_et_al_2023_PopSignASL_v1
  volume = {36},
  year = {2023}
 }
+
+@inproceedings{Cheng2023CiCoSignLanguageRetrieval,
+  author={Cheng, Yiting and Wei, Fangyun and Bao, Jianmin and Chen, Dong and Zhang, Wenqiang},
+  booktitle={2023 IEEE/CVF Conference on Computer Vision and Pattern Recognition (CVPR)}, 
+  title={CiCo: Domain-Aware Sign Language Retrieval via Cross-Lingual Contrastive Learning},
```
Comment (AmitMY) [inline src/references.bib:3285]: {CiCo}

File: src/index.md [hunk 9]
```diff
@@ -901,17 +901,34 @@ TODO
...
 They pretrain a [Sign Language Recognition model](#pose-to-gloss) on a subset of the VGT corpus [@dataset:herreweghe2015VGTCorpus] to embed sign inputs.
 Once the encoder is trained, they use it to generate embeddings for all dictionary signs.
 When a user submits a query video, the system compares the input embeddings with those of the dictionary entries using Euclidean distance.
 Tests on a [proof-of-concept Flemish Sign Language dictionary](https://github.com/m-decoster/VGT-SL-Dictionary) show that the system can successfully retrieve a limited vocabulary of signs, including some not in the training set.
-<!-- TODO: add VGT Corpus (dataset:herreweghe2015VGTCorpus) to list of datasets -->
+
+@Cheng2023CiCoSignLanguageRetrieval introduce a video-to-text and text-to-video retrieval method using cross-lingual contrastive learning.
+Inspired by transfer learning from sign-spotting/segmenting models [@jui-etal-2022-machine;@Duarte2022SignVideoRetrivalWithTextQueries], the authors employ a domain-agnostic I3D encoder, pretrained on large-scale sign language datasets  for the sign-spotting task [@Varol2021ReadAndAttend].
```
Comment (AmitMY) [inline src/index.md:924]: "segmenting" -> "segmentation"

Comment (AmitMY) [inline src/index.md:924]: datasets (double sapce) for

File: src/index.md [hunk 10]
```diff
@@ -901,17 +901,34 @@ TODO
...
+On target datasets with continuous signing videos, they use this model with a sliding window to identify high confidence predictions, which are then used to finetune a "domain-aware" sign-spotting encoder.
+The combined encoders pre-extract features from sign language videos.
```
Comment (AmitMY) [inline src/index.md:926]: how are they "combined"? do you mean "both encoders were used to ..."
also in that case, i don't understand which models are "combined" or "both"

File: src/index.md [hunk 11]
```diff
@@ -901,17 +901,34 @@ TODO
...
+Cross-lingual contrastive learning [@Radford2021LearningTV] is applied to align feature-text pairs within a shared embedding space.
```
Comment (AmitMY) [inline src/index.md:927]: should be merged with the previous sentence. these two fragments don't make sense as individual sentences

File: src/index.md [hunk 12]
```diff
@@ -901,17 +901,34 @@ TODO
...
+Evaluations on How2Sign [@dataset:duarte2020how2sign] and RWTH-PHOENIX-Weather 2014T datasets [@cihan2018neural] demonstrate significant improvment over the previous state of the art method [@Duarte2022SignVideoRetrivalWithTextQueries].
```
Comment (AmitMY) [inline src/index.md:928]: on "the" .... datasets

Comment (AmitMY) [reply src/index.md:928]: "significant" tends to be misused. do they show significance testing? if not, remove that word

**Review by AmitMY — CHANGES_REQUESTED** [review]

File: src/index.md (hunk 7 above)
Comment (cleong110) [reply src/index.md:904]: good catch!

Comment (cleong110) [reply src/index.md:904]: Really ought to pull out my[ rubber duck](https://en.wikipedia.org/wiki/Rubber_duck_debugging) for this.

File: src/index.md (hunk 12 above)
Comment (cleong110) [reply src/index.md:928]: I could have sworn I put "the" in. 

Significance: based on my opinion/eyeballed it. Recall@1 of 30ish to about 50ish for example seemed like a big bump. I didn't put much thought into it though, thanks for calling it out.

Comment (cleong110) [reply src/index.md:928]: What actually say:  "outperforms the pioneering method by large margins on various dataset"

File: src/index.md (hunk 10 above)
Comment (cleong110) [reply src/index.md:926]: Weighted combination of the outputs I believe. 

image
image
> Our final sign encoder H(·), as shown
in Figure 3a, is a weighted combination of hξ (·) and hθ (·)
with a trade-off hyper-parameter α

Comment (cleong110) [reply src/index.md:926]: image

**Comment by cleong110** [comment]
OK, took another crack at CiCo summary

File: src/index.md [hunk 13]
```diff
@@ -888,16 +888,36 @@ TODO
...
+Inspired by transfer learning from sign-spotting/segmentation models [@jui-etal-2022-machine;@Duarte2022SignVideoRetrivalWithTextQueries], the authors employ a "domain-agnostic" I3D encoder, pretrained on large-scale sign language datasets for the sign-spotting task [@Varol2021ReadAndAttend].
+On target datasets with continuous signing videos, they use this model with a sliding window to identify high confidence predictions, which are then used to finetune a "domain-aware" sign-spotting encoder.
+The two encoders each pre-extract features from videos, which are then fused via a weighted sum.
+Cross-lingual contrastive learning [@Radford2021LearningTV] is then applied to align the extracted features with paired texts within a shared embedding space.
+This allows the calculation of similarity scores between text and video embeddings, and thus retrieval in either direction.
+Evaluations on How2Sign [@dataset:duarte2020how2sign] and RWTH-PHOENIX-Weather 2014T datasets [@cihan2018neural] demonstrate  improvment over the previous state of the art method [@Duarte2022SignVideoRetrivalWithTextQueries].
```
Comment (AmitMY) [inline src/index.md:916]: please note, you repeat this issue:
`Evaluations on X and Y datasets` should have a `the`. `Evaluation on the X and Y datasets`

Comment (AmitMY) [reply src/index.md:916]: The word datasets should come after the citation, not before

Comment (AmitMY) [reply src/index.md:916]: double space

Comment (AmitMY) [reply src/index.md:916]: typo: improvement

Comment (AmitMY) [reply src/index.md:916]: `the previous state of the art method` -> `the previous state of the art`

**Review by AmitMY — CHANGES_REQUESTED** [review]
just one sentence with issues

File: src/index.md (hunk 13 above)
Comment (cleong110) [reply src/index.md:916]: Re: "the previous state of the art method" I was trying to get across that there was only one single paper they compared with. It seems to be a new thing

Comment (cleong110) [reply src/index.md:916]: > please note, you repeat this issue: `Evaluations on X and Y datasets` should have a `the`. `Evaluation on the X and Y datasets`

Ah yes, time to add this to my checklist

Comment (cleong110) [reply src/index.md:916]: Also, going to give https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker a try

**Review by AmitMY — APPROVED** [review]
